test/suite/test_backup29.py

```diff
@@ -0,0 +1,141 @@
+
+
+    def test_backup29(self):
+        os.mkdir(self.dir)
+
+        # Create and populate the table.
+        file_uri = 'file:' + self.uri + '.wt'
+        file2_uri = 'file:' + self.uri2 + '.wt'
+        table_uri = 'table:' + self.uri
+        table2_uri = 'table:' + self.uri2
+        self.session.create(table_uri, self.create_config)
+        self.session.create(table2_uri, self.create_config)
+        c = self.session.open_cursor(table_uri)
+        c2 = self.session.open_cursor(table2_uri)
+        # Only add a few entries.
+        self.pr("Write: " + str(self.few) + " initial data items")
+        for i in range(1, self.few):
+            val = str(i) + self.value_base
+            c[i] = val
+            c2[i] = val
+        self.session.checkpoint()
+
+        # Take the initial full backup for incremental.
+        config = 'incremental=(enabled,granularity=4k,this_id="ID1")'
+        bkup_c = self.session.open_cursor('backup:', None, config)
+        self.take_full_backup(self.dir, bkup_c)
+        bkup_c.close()
+
+        # Add a lot more data to both tables to generate a filled in block mod bitmap.
+        last_i = self.few
+        self.pr("Write: " + str(self.nentries) + " additional data items")
+        for i in range(self.few, self.nentries):
+            val = str(i) + self.value_base
+            c[i] = val
+            c2[i] = val
+        last_i = self.nentries
+        c.close()
+        c2.close()
+        self.session.checkpoint()
+        # Get the block mod bitmap from the file URI.
+        orig_id1blocks = self.parse_blkmods(file2_uri)
+        self.pr("CLOSE and REOPEN conn")
+        self.reopen_conn()
+        self.pr("Reopened conn")
+
+        # After reopening we want to open both tables, but only modify one of them for
+        # the first checkpoint. Then modify both tables, checkpoint, and then check the
+        # that the block mod bitmap remains correct.
+        c = self.session.open_cursor(table_uri)
+        c2 = self.session.open_cursor(table2_uri)
+
+        # Change one table and checkpoint.
+        self.pr("Update only table 1: " + str(last_i))
+        far_key = self.nentries * 2
+        val = str(far_key) + self.value_base
+        c[far_key] = val
+        self.session.checkpoint()
+        # Now change the other table and checkpoint again.
+        val = str(last_i) + self.value_base
+        self.pr("Update second table: " + str(last_i))
+        c2[last_i] = val
+        self.session.checkpoint()
+        new_id1blocks = self.parse_blkmods(file2_uri)
+
+        c.close()
+        c2.close()
+
+        # Compare the bitmaps from the metadata. Once a bit is set, it should never
+        # be cleared. But new bits could be set. So the check is only: if the original
+        # bitmap has a bit set then the current bitmap must be set for that bit also. 
+        for orig, new in zip(orig_id1blocks, new_id1blocks):
+            if orig != '0':
+                self.assertTrue(new != '0')
```

Comment:
If I'm reading this correctly, we're comparing the bitmaps a nibble at a time? So the test is basically saying that if the original has any of the four bits in a nibble set, then the current shouldn't have all four clear.

Two thoughts: 

1. It seems it would be better to actually compare bit-by-bit -- it would catch more error cases.
2. If not, the comment should say something about the fact that we're simplifying the comparison by just looking at 4 bits at a time.

Reply:
@keitharnoldsmith take a look now.